handlers/status_handler.cc

```diff
@@ -21,10 +21,11 @@ RequestHandler::Status StatusHandler::HandleRequest(const Request& request,
 
   ServerStatus::Status status = status_->GetStatus();
 
-
+  status.stsMx->lock();
```

Comment:
Why are you locking the status here? You lock it and then don't use the resource.